docs/fundamentals/code-analysis/rules/snippets/csharp/all-rules/ca2215.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+
+namespace ca2215
+{
+    public class TypeA : IDisposable
+    {
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                // Dispose managed resources
+            }
+
+            // Free native resources
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        // Disposable types implement a finalizer.
+        ~TypeA()
+        {
+            Dispose(false);
+        }
+    }
+
+    public class TypeB : TypeA
```

Comment:
This is new code. Please check. I did verify that it satisfies rule CA2215 but not sure if it's otherwise correct.